File: leetcode/go/1518.water-bottles.go
func numWaterBottles(numBottles int, numExchange int) int {
    ans := 0
    for numBottles >= numExchange {
        x := numBottles/numExchange
        ans += x * numExchange
        numBottles -= (x * numExchange - x)
    }
    ans += numBottles
    return ans
}

File: leetcode/rust-leetcode/src/1518.water-bottles.rs
impl Solution {
    pub fn num_water_bottles(mut num_bottles: i32, mut num_exchange: i32) -> i32 {
        let mut ans = 0;
        while num_bottles >= num_exchange {
            let k = num_bottles/num_exchange;
            ans += k * num_exchange;
            num_bottles = num_bottles - k * num_exchange + k;
        }
        ans += num_bottles;
        ans
    }
}

